src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,18 +3,47 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int INITIAL_SIZE = 10;
+    private final V[] values;
+    private final K[] keys;
+    private int size;
+
+    public StorageImpl() {
+        values = (V[]) new Object[INITIAL_SIZE];
+        keys = (K[]) new Object[INITIAL_SIZE];
+    }
 
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < size; i++) {
+            if (keys[i] != null && keys[i].equals(key)) {
+                keys[i] = key;
+                values[i] = value;
```

Comment:
```suggestion
                values[i] = value;
```
keys are the same if we're inside of if, no need to reassign them